sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -303,16 +303,6 @@ esimd_lsr(T1 src0, T2 src1, int flag = saturation_off) {
   return Result[0];
 }
 
-template <typename T0, typename T1, typename T2>
-ESIMD_NODEBUG ESIMD_INLINE typename sycl::detail::enable_if_t<
-    detail::is_esimd_scalar<T1>::value && detail::is_esimd_vector<T2>::value &&
-        std::is_integral<T0>::value && std::is_integral<T1>::value &&
-        std::is_integral<T2>::value,
```

Comment:
This looks like a bug. It should be std::is_integral\<T2::element_type\>::value

Reply:
I can't understand the value of this overload. Looks like it is an alias for the first version, where we have the first argument a vector, and the second argument a scalar. Here we just swap the order of arguments but to me, it really seems confusing. Unless I'm missing something.

Reply:
I think this was ported to match existing CM math function. IIRC there was customer request to add this convenience API so that when the parameters are specified in different order we will swap it transparently and it's still expected to work.

Reply:
Still looks confusing :)
`esimd_lsr(scalar, vector)` will actually do `vector >> scalar`. I don't see much value in this API. Readers of the code will probably get confused over the order of arguments.

Reply:
I see your confusion. This was added per previous customer request for programming convenience. I'm fine with removing it  if it's confusing for ESIMD user. This could be done in kernel or utility file. I don't view it as a blocking issue.